### PR TITLE
Update fetcher to support additional methods

### PR DESCRIPTION
**Pull request by colinking**
This PR adds support for the other main HTTP methods.

It also updates the type of query params on `.get()` to allow passing through `null/undefined` (used by Views). It also updates the `headers` parameter to match for consistency.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #269** <a href="https://app.graphite.dev/github/pr/airplanedev/node-sdk/269" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/node-sdk/269?utm_source=stack-comment).

File: packages/lib/internal/fetcher.ts [hunk 1]
```diff
@@ -100,15 +136,18 @@ export class Fetcher {
       }
     }
 
+    const headers: Record<string, string> = {};
+    for (const [k, v] of Object.entries(opts.headers ?? {})) {
+      if (v) {
+        headers[k] = v;
+      }
```
Comment (ericliuche) [inline packages/lib/internal/fetcher.ts:143]: will this exclude `0`, `""`, etc?

Comment (colinking) [reply packages/lib/internal/fetcher.ts:143]: It excludes the empty string, which is intentional. The type is `string`, so it won't receive numbers.

Comment (colinking) [reply packages/lib/internal/fetcher.ts:144]: Added a comment in code.

**Review by ericliuche — APPROVED** [review]

**Pull request merged**